app/controllers/api/v2_alpha/notifications_controller.rb

```diff
@@ -12,10 +12,26 @@ def index
     with_read_replica do
       @notifications = load_notifications
       @group_metadata = load_group_metadata
+      @grouped_notifications = load_grouped_notifications
       @relationships = StatusRelationshipsPresenter.new(target_statuses_from_notifications, current_user&.account_id)
+
+      # Preload associations to avoid N+1s
+      ActiveRecord::Associations::Preloader.new(records: @grouped_notifications.flat_map(&:sample_accounts), associations: [:account_stat, { user: :role }]).call
```

Comment:
Since `@grouped_notifications.flat_map(&:sample_accounts)` is now used in the OTEL instrumentation as well, it might make sense to cache this in an instance variable.

Reply:
Ah, right, changed!